src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -78,13 +85,80 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String user = userService.getCurrentUser().getEmail();
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
 	
-    String regex = "(https?://\\S+\\.(png|jpg))";
-    String replacement = "<img src=\"$1\" />";
-    String textWithImagesReplaced = text.replaceAll(regex, replacement);
-
+    String textWithImagesReplaced = tagImageURLs(text);
+	
     Message message = new Message(user, textWithImagesReplaced);
     datastore.storeMessage(message);
 
     response.sendRedirect("/user-page.html?user=" + user);
   }
+  
```

Comment:
I have added some code for url validation, but there are some problems that I have not been able to fix:
1) https://example1.jpghttps://example2.jpg ->get in image tag as one single item! but it should be two.
2)is https://example.com.com.jpg valid? the code sees it valid, but because I don't know what is valid and what is not I have problem checking the results!
I need some help to clean and improve this code, I appreciate any help ! Thank you so much Ginelle :)

Reply:
one thing that happen similar tow problem 1 when I run the code and I give a String like : 
"Hellohttps://example.jpgBye" it works and gives Hello<img https://.../>Bye
But if I pass it:
"Hellohttps://example.jpgByehttps://example.jpg", similar to number one it does not separate urls and gives Hello<img src = "https://...Byehttps://...>
Unless I add a white space between the two urls, it does not separate them

Reply:
That is awesome! adding ? after S*  works! I tried many different things yesterday and none of them worked! Thank you soooooooooooooo much :)